venonactl/pkg/plugins/helper.go

```diff
@@ -130,3 +149,64 @@ func getKubeObjectsFromTempalte(values map[string]interface{}, pattern string, l
 	templatesMap := templates.TemplatesMap()
 	return KubeObjectsFromTemplates(templatesMap, values, pattern, logger)
 }
+
+func ensureClusterRequirements(client *kubernetes.Clientset, req validationRequest) (validationResult, error) {
+	result := validationResult{}
+	nodes, err := client.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return result, err
+	}
+	if nodes == nil {
+		return result, errors.New("Nodes not found")
+	}
+
+	result.isValid = true
+	if len(nodes.Items) == 0 {
+		result.message = append(result.message, "No nodes in cluster")
+		result.isValid = false
+	}
+	for _, n := range nodes.Items {
+		res := testNode(n, req)
+		if len(res) > 0 {
+			result.message = append(result.message, res...)
+			result.isValid = false
+		}
+	}
+	if !req.strictMode {
```

Comment:
move this to the beginning of the function, no need to perform the check on non strict mode